specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json

```diff
@@ -0,0 +1,421 @@
+      "put": {
+        "tags": [
+          "ConfigurationProfileAssignmentIntents"
+        ],
+        "operationId": "ConfigurationProfileAssignmentIntents_CreateOrUpdate",
+        "description": "Creates an association between a VM and Automanage configuration profile assignment intent",
+        "x-ms-examples": {
+          "Create or update configuration profile assignment intent": {
+            "$ref": "./examples/createOrUpdateConfigurationProfileAssignmentIntent.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "configurationProfileAssignmentIntentName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Name of the configuration profile assignment intent. Only default is supported."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ConfigurationProfileAssignmentIntent"
+            },
+            "description": "Parameters supplied to the create or update configuration profile assignment intent."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/VmNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Created. The Automanage configuration profile assignment intent for the VM was created successfully.",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationProfileAssignmentIntent"
+            }
+          },
+          "200": {
+            "description": "OK. The configuration profile assignment intent for the VM was updated successfully.",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationProfileAssignmentIntent"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": false,
+        "produces": [
+          "application/json"
+        ],
+        "consumes": [
+          "application/json"
+        ]
+      },
+      "get": {
```

Comment:
put and get but no delete?  How does someone remove an intent that they created?

Reply:
When an configurationProfileAssignments is deleted we will delete the related configurationProfileAssignmentIntents too. That is why we did not implement the delete.

Reply:
Can we get your manager's comment regarding this design choice please? You are violating our guideline, see: [Incosistent or Incomplete Crud](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

---
In reply to: [603708234](https://github.com/Azure/azure-rest-api-specs/pull/13556#discussion_r603708234) [](ancestors = 603708234)

Reply:
Please scan comments in https://stackoverflow.microsoft.com/questions/216549

Reply:
Got it, if this proxy resource, you are good, thanks!

---
In reply to: [605165563](https://github.com/Azure/azure-rest-api-specs/pull/13556#discussion_r605165563) [](ancestors = 605165563)